roles/keystone/templates/etc/keystone/policy.json

```diff
@@ -178,14 +171,14 @@
     "identity:delete_protocol": "rule:admin_required",
 
     "identity:create_mapping": "rule:admin_required",
-    "identity:get_mapping": "rule:admin_required",
-    "identity:list_mappings": "rule:admin_required",
+    "identity:get_mapping": "rule:admin_or_cloudadmin",
+    "identity:list_mappings": "rule:admin_or_cloudadmin",
     "identity:delete_mapping": "rule:admin_required",
-    "identity:update_mapping": "rule:admin_required",
+    "identity:update_mapping": "rule:admin_or_cloudadmin",
 
     "identity:create_service_provider": "rule:admin_required",
-    "identity:list_service_providers": "rule:admin_required",
-    "identity:get_service_provider": "rule:admin_required",
+    "identity:list_service_providers": "rule:admin_or_cloudadmin_or_projectadmin_or_member",
+    "identity:get_service_provider": "rule:admin_or_cloudadmin",
```

Comment:
everyone needs to be able to list the service providers but only admin and cloud admin need to know the details for the service providers. For now this looks good, if we need to open up more or close it down we will re-evaluate after this goes in.